src/dex/dexalot/dexalot.ts

```diff
@@ -0,0 +1,935 @@
+
+      const pairData = await this.getPairData(
+        normalizedSrcToken,
+        normalizedDestToken,
+      );
+      if (!pairData) {
+        return null;
+      }
+
+      const priceMap = await this.getCachedPrices();
+      if (!priceMap) {
+        return null;
+      }
+
+      const pairKey = `${pairData.base}/${pairData.quote}`.toLowerCase();
+      if (!(pairKey in priceMap)) {
+        return null;
+      }
+
+      const priceData = priceMap[pairKey];
+      const baseToken = pairData.isSrcBase
+        ? normalizedSrcToken
+        : normalizedDestToken;
+      const quoteToken = pairData.isSrcBase
+        ? normalizedDestToken
+        : normalizedSrcToken;
+
+      // convert from swap to clob side
+      let orderbook = priceData.asks;
+      let clobSide = ClobSide.BID;
+      if (
+        (side === SwapSide.SELL && pairData.isSrcBase) ||
+        (side === SwapSide.BUY && !pairData.isSrcBase)
+      ) {
+        orderbook = priceData.bids;
+        clobSide = ClobSide.ASK;
+      }
+      if (orderbook.length === 0) {
+        throw new Error(`Empty orderbook for ${pairKey}`);
+      }
+
+      const prices = this.calculateOrderPrice(
+        amounts,
+        orderbook,
+        baseToken,
+        quoteToken,
+        clobSide,
+      );
+      const outDecimals =
+        clobSide === ClobSide.BID ? baseToken.decimals : quoteToken.decimals;
+      const poolIdentifier = this.getIdentifier(pairData.base, pairData.quote);
+
+      return [
+        {
+          prices,
+          unit: BigInt(outDecimals),
+          data: {},
+          poolIdentifier: poolIdentifier,
+          exchange: this.dexKey,
+          gasCost: DEXALOT_GAS_COST,
+          poolAddresses: [this.mainnetRFQAddress],
+        },
+      ];
+    } catch (e: unknown) {
+      this.logger.error(
+        `Error_getPricesVolume ${srcToken.symbol || srcToken.address}, ${
+          destToken.symbol || destToken.address
+        }, ${side}:`,
+        e,
+      );
+      return null;
+    }
+  }
+
+  generateRFQError(errorStr: string, swapIdentifier: string) {
+    const message = `${this.dexKey}-${this.network}: Failed to fetch RFQ for ${swapIdentifier}. ${errorStr}`;
+    this.logger.warn(message);
+    throw new DexalotRfqError(message);
+  }
+
+  async preProcessTransaction(
+    optimalSwapExchange: OptimalSwapExchange<DexalotData>,
+    srcToken: Token,
+    destToken: Token,
+    side: SwapSide,
+    options: PreprocessTransactionOptions,
+  ): Promise<[OptimalSwapExchange<DexalotData>, ExchangeTxInfo]> {
+    if (await this.isBlacklisted(options.txOrigin)) {
+      this.logger.warn(
+        `${this.dexKey}-${this.network}: blacklisted TX Origin address '${options.txOrigin}' trying to build a transaction. Bailing...`,
+      );
+      throw new Error(
+        `${this.dexKey}-${
+          this.network
+        }: user=${options.txOrigin.toLowerCase()} is blacklisted`,
+      );
+    }
+    if (await this.isRateLimited(options.txOrigin)) {
```

Comment:
If we have notion of rate limiting on Dexalot, we might want to adjust API to act accordingly and hide Dexalot for user, if he is rate limited. Otherwise, I am afraid we might have quite a lot failures.